src/msg/async/dpdk/DPDK.h

```diff
@@ -541,7 +541,7 @@ class DPDKQueuePair {
   uint32_t send(circular_buffer<Packet>& pb) {
     // Zero-copy send
     return _send(pb, [&] (Packet& p) {
-      return tx_buf::from_packet_zc(cct, p, *this);
+      return tx_buf::from_packet_copy(p, *this);
```

Comment:
could you expand on "not realy zero copy" a little bit? `DPDKQueuePair::send()` is the only caller of `tx_buf::from_packet_zc()` . if the latter is a buggy implementation, we might need to fix it instead of using an alternative which does the deep copy.

Reply:
The prerequisite for zero copy is that rte_malloc is used. Currently, packets are sent from various mallocs in buffer::raw. Therefore, zero copy cannot be used directly.
In the set_one_data_buf function, it is return copy_one_data_buf(qp, m, va, buf_len)  , instead of going through the address translation. so this not realy zero copy.